stdlib/src/lib.rs

```diff
@@ -37,7 +42,12 @@ impl StdLibrary {
 
 impl Default for StdLibrary {
     fn default() -> Self {
-        Self(Library::read_from_bytes(Self::SERIALIZED).expect("failed to deserialize stdlib"))
+        static STDLIB: LazyLock<StdLibrary> = LazyLock::new(|| {
+            let contents =
+                Library::read_from_bytes(StdLibrary::SERIALIZED).expect("failed to read std masl!");
```

Comment:
nit: I would probably just say `Self::SERIALIZED` here.

Reply:
Compiler does not allow it due to the static:
```
error[E0401]: can't use `Self` from outer item
  --> stdlib/src/lib.rs:47:42
   |
43 | impl Default for StdLibrary {
   | ---- `Self` type implicitly declared here, by this `impl`
...
47 |                 Library::read_from_bytes(Self::SERIALIZED).expect("failed to read std masl!");
   |                                          ^^^^^^^^^^^^^^^^
   |                                          |
   |                                          use of `Self` from outer item
   |                                          refer to the type directly here instead
   |
   = note: a `static` is a separate item from the item that contains it
```

Also note that this is required to avoid another warning:
```rust
StdLibrary(contents)
```
Warning if `Self` is used:
```
can't reference `Self` constructor from outer item
this was previously accepted by the compiler but is being phased out; it will become a hard error in a future release!
for more information, see issue #124186 <https://github.com/rust-lang/rust/issues/124186>
`#[warn(self_constructor_from_outer_item)]` on by defaultrustc[Click for full compiler diagnostic](rust-analyzer-diagnostics-view:/diagnostic%20message%20%5B0%5D?0#file%3A%2F%2F%2FUsers%2Fsergeradinovich%2FSource%2Fsergerad%2Fmiden-vm%2Fstdlib%2Fsrc%2Flib.rs)
lib.rs(43, 1): the inner item doesn't inherit generics from this impl, so `Self` is invalid to reference
lib.rs(48, 13): replace `Self` with the actual type: `StdLibrary`
```

Reply:
Ah - interesting! Let's keep the way you have it then.